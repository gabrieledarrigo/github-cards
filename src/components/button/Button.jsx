import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.css';

export const ButtonType = {
  BUTTON: 'button',
  SUBMIT: 'submit',
};

export default function Button({
  type,
  children,
  ...rest
}) {
  return (
    // eslint-disable-next-line react/button-has-type
    <button
      className={styles.button}
      type={type}
      {...rest}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.oneOf([
    ButtonType.BUTTON,
    ButtonType.SUBMIT,
  ]),
  children: PropTypes.node.isRequired,
};

Button.defaultProps = {
  type: ButtonType.BUTTON,
};
