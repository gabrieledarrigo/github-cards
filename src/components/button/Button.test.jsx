import React from 'react';
import renderer from 'react-test-renderer';
import Button, { ButtonType } from './Button';

describe('Button', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Button type={ButtonType.BUTTON}>
        A text
      </Button>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
