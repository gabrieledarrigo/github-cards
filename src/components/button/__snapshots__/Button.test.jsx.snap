// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="button"
  type="button"
>
  A text
</button>
`;
